lib/puppet/pops/pcore.rb

```diff
@@ -42,6 +42,15 @@ def self.init(loader, ir, for_agent)
     if Puppet[:tasks]
       require_relative 'types/task'
       require_relative 'types/execution_result'
+
+      add_object_type('TargetNode', <<-PUPPET, loader)
```

Comment:
The name change from the agreed `Node` to `TargetNode` was not signed off on the ticket. I am not against the term 'target node' as I think it is clearer that you are talking about the technical reference to a node on which we are going to execute tasks. But need @adreyer to sign off on this. (Technically we can use the name `Node` since we do not have such a resource data type exposed in the type system).

Reply:
I discovered that switching to `Node` will require lots of tests to change since many of them causes collisions due to various ways of initializing the runtime before enabling `Puppet[:tasks]`, which proves my point that `Node` is not an ideal name since it's hard to figure out all places where such collisions may take place.

I also feel that since the name `Node` is an established "thing" in Puppet, and even a keyword in the parser, it shouldn't be considered as the name for a new data type. It's confusing enough as it is.

Reply:
@adreyer , your input is required here before this PR can be merged.

Reply:
I don't see `Target` as clashing with anything else in general, so ok to use from that perspective. IMO - taken out of context it seems a bit too generic. @adreyer  are you sure you want just `Target`? The first time a user may see it is when they read the source of a plan, and they see something like `plan foo(Array[Target] $targets) { }` - and you then have to think "what is a Target for a plan?"... whereas `TargetNode` tells you that this is about a node.